skiko/src/jvmMain/kotlin/org/jetbrains/skiko/redrawer/MetalRedrawer.kt

```diff
@@ -28,8 +32,10 @@ internal class MetalRedrawer(
     private val windowHandle = layer.windowHandle
 
     private val frameDispatcher = FrameDispatcher(Dispatchers.Swing) {
-        update(System.nanoTime())
-        draw()
+        if (layer.isShowing) {
```

Comment:
How expensive is isShowing check?

Reply:
It checks a boolean property recursively through all parents.

With 1000 deeply nested components (30 level-deep) we have 0.5ms.

Let's cached it then.